R/get.R

```diff
@@ -91,10 +98,47 @@ docdb_get.src_mongo <- function(src, key, limit = NULL, ...) {
   if (!is.null(limit)) return(src$con$iterate(limit = limit)$page())
   dump <- tempfile()
   src$con$export(file(dump))
-  # remove first column, a mongodb identifier
-  jsonlite::stream_in(file(dump), verbose = FALSE)[,-1]
+  jsonlite::stream_in(file(dump), verbose = FALSE)
+}
+
+#' @export
+docdb_get.src_sqlite <- function(src, key, limit = NULL, ...) {
+  
+  assert(key, "character")
+  assert(limit, "integer")
+  
+  # arguments for call
+  statement <- paste0(
+    # _id is included into json for use with jsonlite::stream_in
+    "SELECT '{\"_id\": \"' || _id || '\", ' || substr(json, 2) ", 
+    "FROM ", key, " ;")
+  
+  # set limit if not null
+  n <- -1L
+  if (!is.null(limit)) n <- limit
+  
+  # temporary file; note this cannot be deleted
+  # because it is streamed into the return value
+  dump <- tempfile()
+  
+  # get data, write to file in ndjson format
+  cat(stats::na.omit(unlist(
```

Comment:
I wonder if this is necessary to write to a file first? I haven't looked at this in detail so I may be wrong here. I guess `jsonlite::stream_in` only can accept a file/url/gzcon, so maybe file is the only way

Reply:
In the code comments just before `jsonlite::stream_in()`, I cite the jsonlite developer's reason for this function: parsing huge JSON strings is difficult and inefficient (see also [https://rdrr.io/cran/jsonlite/man/stream_in.html#heading-3](https://rdrr.io/cran/jsonlite/man/stream_in.html#heading-3). 

In addition, `nodbi::docdb_get.src_mongo()` already used `jsonlite::stream_in()` so that I considered this a relevant option also for `nodbi::docdb_get.src_sqlite()`. The file in a temporary space of the local file system greatly simplifies implementation. 

Also it comes with a significant speed improvement, as can be seen now for `nodbi::docdb_create.src_sqlite()` and this addresses the issue raised above.

Reply:
ok, thanks